### DISCUSSION TITLE
Redirect when Fetch has no Results

### DISCUSSION THREAD

**Discussion by paleacci** [nuxt/nuxt#17770] [Questions]
Hello,

how is it possible in nuxt 3 to show a 404 error page when the my fetch result is empty?

I have already tried a few things, but unfortunately without success. Meanwhile my code looks like this because nothing else worked. But I'm sure it can be done better.

 `async setup() {
        const route = useRoute();
        const router = useRouter();

        const item = await $fetch('/api/load-some-data', { params: {
            someparam:1 route?.params?.params[1],
            someparam2: route?.params?.params[0]
        }});

        if(!item) {
            await router.push('/');
        }

        return {
            item
        }
 }`

**Comment by guzz** [comment]
You can use the createError function from 'h3' and the throwError from nuxt, like this:

throwError(createError({
    statusCode: 404,
    message: 'Error message'
  }))

**Comment by findtoni** [comment]
```
const { data: item, error } = await useFetch('/api/load-some-data', { 
  params: {
      someparam:1 route?.params?.params[1],
      someparam2: route?.params?.params[0]
  }
});

!item.value.length >=1 || error.value ? router.push('/') : '';
return { item };
```
